contrib/build-push-ceph-container-imgs.sh

```diff
@@ -44,22 +44,63 @@ function enable_experimental_docker_cli {
   fi
 }
 
+function compare_docker_hub_and_github_tags {
+  # build an array with the list of tags from github
+  for tag_github in $(git ls-remote --tags | grep -oE 'v[3-9].[0-9]*.[0-9]*$' | sort -t. -k 1,1n -k 2,2n -k 3,3n -k 4,4n); do
+    tags_github_array+=("$tag_github")
+  done
+
+  # download cn to list docker hub images, it's easier than building the logic in bash...
+  # and cn is only 10MB so it doesn't hurt
+  curl -L https://github.com/ceph/cn/releases/download/v1.7.0/cn-v1.7.0-cc5ad52-linux-amd64 -o cn
+  chmod +x cn
+
+  # build an array with the list of tag from docker hub
+  tags_docker_hub_unsorted="$(./cn image ls -a | grep -oE 'v[3-9].[0-9]*.[0-9]' | uniq)"
+  tags_docker_hub_sorted="$(echo "$tags_docker_hub_unsorted" | sort -t. -k 1,1n -k 2,2n -k 3,3n -k 4,4n)"
+  for tag_docker_hub in $tags_docker_hub_sorted; do
+    tags_docker_hub_array+=("$tag_docker_hub")
+  done
+
+  # we now look into each array and find a possible missing tag
+  # the idea is to find if a tag present on github is not present on docker hub
+  for i in "${tags_github_array[@]}"; do
```

Comment:
I suggest a shorter code

  for i in "${tags_github_array[@]}"; do
    echo "${tags_docker_hub_array[@]}" |grep -q "$i" || tag_to_build+=("$i")
  done